conan/tools/scm/__init__.py

```diff
@@ -1,2 +1,200 @@
 from conan.tools.scm.git import Git
-from conans.model.version import Version
+# develop2 uncomment the next line
+# from conans.model.version import Version
+
+
+# ######## DEVELOP2: Delete all the contents from here, this is copy-paste from the Version model
+# ######## at conans.model.version (for an easier migration)
+from functools import total_ordering
+from conans.errors import ConanException
+
+
+@total_ordering
+class _VersionItem:
+    """ a single "digit" in a version, like X.Y.Z all X and Y and Z are VersionItems
+    They can be int or strings
+    """
+    def __init__(self, item):
+        try:
+            self._v = int(item)
+        except ValueError:
+            self._v = item
+
+    @property
+    def value(self):
+        return self._v
+
+    def __str__(self):
+        return str(self._v)
+
+    def __add__(self, other):
+        # necessary for the "bump()" functionality. Other aritmetic operations are missing
+        return self._v + other
+
+    def __eq__(self, other):
+        if not isinstance(other, _VersionItem):
+            other = _VersionItem(other)
+        return self._v == other._v
+
+    def __hash__(self):
+        return hash(self._v)
+
+    def __lt__(self, other):
+        """
+        @type other: _VersionItem
+        """
+        if not isinstance(other, _VersionItem):
+            other = _VersionItem(other)
+        try:
+            return self._v < other._v
+        except TypeError:
+            return str(self._v) < str(other._v)
+
+
+@total_ordering
+class Version:
```

Comment:
Could you please add `as_list`, so it could be minimal compatible with what we have in Conan Center Index now.

Reply:
I wouldn't like that, I think that ``as_list`` is an ugly interface that if we do that it will "infect" Conan 2.0.
If necessary, please let us know the use cases of such method, and we can think about what is the right interface for it.

Reply:
I still prefer the legacy, as described in this PR. The only thing is re-editing recipes again in CCI. Also, we will need to create some `if conan_version else` because the different behavior from 1.49 (i guess) til 1.52

Reply:
I haven't found occurrences of ``as_list`` in conan-center-index (only one method in openssl recipe), am I missing something?

Reply:
Boost recipe: https://github.com/conan-io/conan-center-index/blob/master/recipes/boost/all/conanfile.py#L1461

If you are running GH search, forget, it's useless most of time.

Reply:
No, I was running my local clone. But it wasn't updated. It seems it is only the boost recipe, and it was very recently added.

It seems what the boost recipe could be done more idiomatically with ``version.major`` and ``version.minor`` isn't it?

Reply:
Furthermore, ``as_list`` is not a documented/public method at all, shouldn't be used in any of the ``Version`` implementations? While ``major`` and ``minor`` are. Not sure why they used that in the boost recipe.

Reply:
The problem was, Boost recipe need to compare some major and minor versions, but the new scm.Version does not support semver, so it return "1.2" for Version.minor(fill=False). Of course, the previous behavior, with semver, was much better. I gonna revert those changes and keep aligned with this new PR. Thanks.

Reply:
Not sure what you mean. This new ``Version`` in this PR supports semver like comparison between versions, no problem with that. How is that related to the ``as_list``  usage, which is done to just extract the major, minor and patch to compose some strings (suffixes) in the boost recipe?

Reply:
- The new Version changed major, minor and patch from property to method
- When you call `conan.tools.scm.Version('1.2.3').minor()` it returns `1.2.Z`
- When you call `conan.tools.scm.Version('1.2.3').minor(fill=False)` it returns `1.2`
- But still, I only want the minor, the `2`.
- As alternative, I can consume from `conan.tools.scm.Version('1.2.3')[1]`

Anyway, this new PR fixes the case, so we can use Version.major as a property again.

Reply:
When I said `The new Version`, I meant which is production right now and running on CCI, not this PR. This PR is correct, it follows the previous and expected behavior.